libtcmu_log.c

```diff
@@ -42,40 +77,93 @@ unsigned int tcmu_get_log_level(void)
 	return tcmu_log_level;
 }
 
-/* covert log level from tcmu config to syslog */
 void tcmu_set_log_level(int level)
 {
-	tcmu_log_level = tcmu_log_level_conf_to_syslog(level);
+	tcmu_log_level = to_syslog_level(level);
 }
 
-void tcmu_log_open_syslog(const char *ident, int option, int facility)
+static void open_syslog(const char *ident, int option, int facility)
 {
-	const char *id = TCMU_IDENT;
+#define ID_MAX_LEN 16
+	char id[ID_MAX_LEN] = {0}, path[128];
+	int fd, len;
 
-	if (ident)
-		id = ident;
+	if (!ident) {
+		sprintf(path, "/proc/%d/comm", getpid());
+		fd = open(path, O_RDONLY);
+			if (fd < 0)
+				return;
+		len = read(fd, id, ID_MAX_LEN);
+		if (len < 0) {
+			close(fd);
+			return;
+		}
+		close(fd);
+	} else {
+		strncpy(id, ident, ID_MAX_LEN);
+	}
 
```

Comment:
We make the id lenth (ID_MAX_LEN + 1), and init id by"char id[ID_MAX_LEN + 1] = {0}",so after strncpy, the last chars will all remain 0.
Quato from man page, "If the length of src is less than n, strncpy() writes additional null bytes to dest to ensure that a total of n bytes are written.".
With above, we can make sure there is a closing '\0' in id.